### ISSUE TITLE
TS module not found until server is restarted

### ISSUE THREAD

**Issue by connor4312** [microsoft/TypeScript#59129]
In VS Code, using the workspace TS version (5.6.0-dev.20240618)

1. Create a new TS file
2. Try to import an existing module
3. Get a "not found" error until you restart the TS language server

![image](https://github.com/user-attachments/assets/6a20126a-d2c7-4d0a-bcba-821bb8a653be)

**Comment by mjbvz** [comment]
Thanks @connor4312 

IN the logs, I can see that `another.ts` ends up in the inferred project instead of the main project:

```
Info 8566 [09:13:01.831] 	FileName: c:/Users/conno/Github/vscode/src/vs/workbench/contrib/testing/common/constants.ts ProjectRootPath: c:/Users/conno/Github/vscode
Info 8566 [09:13:01.831] 		Projects: c:/Users/conno/Github/vscode/src/tsconfig.json
Info 8566 [09:13:01.831] 	FileName: c:/Users/conno/Github/vscode/src/vs/workbench/contrib/testing/common/another.ts ProjectRootPath: c:/Users/conno/Github/vscode
Info 8566 [09:13:01.831] 		Projects: /dev/null/inferredProject1*
```

**Comment by andrewbranch** [comment]
Do you happen to know if this was a recent regression? Did you test on a previous version?

**Comment by connor4312** [comment]
I have seen this happen before, but I think it's been a while (a few releases at least.)

**Comment by bragagia** [comment]
Hey, I just found your issue after fiddling with this problem for weeks now, which was driving me crazy.

After extensive research I found the solution here: https://github.com/microsoft/vscode/issues/161570

TL;DR: Just disable the IntelliCode extensions ;)

**Comment by connor4312** [comment]
Unfortunately I don't have intellicode installed

**Comment by bragagia** [comment]
I deleted my previous comment just after posting because the issue was actually not resolved and the error where not showing during my tests for some other random reason.

But great news, I actually found a new solution which is to disable the enableProjectDiagnostics option. -> https://github.com/microsoft/vscode/issues/168410

settings.json:
```
"typescript.tsserver.experimental.enableProjectDiagnostics": false,
```

That's unfortunate because now only opened files report error, but that's still better than having random errors on every new files.

**Comment by sheetalkamat** [comment]
I think this is fixed by #59625
